pkg/optimizely/cache.go

```diff
@@ -107,14 +132,43 @@ func defaultLoader(
 	conf config.ClientConfig,
 	metricsRegistry *MetricsRegistry,
 	pcFactory func(sdkKey string, options ...sdkconfig.OptionFunc) SyncedConfigManager,
-	bpFactory func(options ...event.BPOptionConfig) *event.BatchEventProcessor) func(sdkKey string) (*OptlyClient, error) {
-	return func(sdkKey string) (*OptlyClient, error) {
+	bpFactory func(options ...event.BPOptionConfig) *event.BatchEventProcessor) func(clientKey string) (*OptlyClient, error) {
+	return func(clientKey string) (*OptlyClient, error) {
+		var sdkKey string
+		var datafileAccessToken string
+		var configManager SyncedConfigManager
+
+		clientKeySplit := strings.Split(clientKey, ":")
+
+		// If there is no : then it is a public datafile and we set sdkKey equal to clientKey
+		if len(clientKeySplit) == 1 {
+			sdkKey = clientKey
+		}
+
+		// If there is a : then it is an authenticated datafile.
+		// First part is the sdkKey.
+		// Second part is the datafileAccessToken
+		if len(clientKeySplit) == 2 {
+			sdkKey = clientKeySplit[0]
+			datafileAccessToken = clientKeySplit[1]
+		}
```

Comment:
Per my previous comment, can we support this feature by allowing the access token to be passed via the existing header as opposed to adding a new header, just to combine it later?